app/models/version.rb

```diff
@@ -34,14 +34,14 @@ def self.preview_version_by_time(time)
   end
 
   def check_version
-    if version.present? && Version.find_by(version: version).nil?
-      errors.add(:version, "#{version} doesn't exist")
+    if number.present? && Version.find_by(number: number).nil?
+      errors.add(:number, "Version number #{number} doesn't exist")
     end
 
     true
   end
 
   def increment_version
-    self.version = (Version.maximum(:version) || 0) + 1 if version.nil?
+    self.number = (Version.maximum(:number) || 0) + 1 if number.nil?
```

Comment:
The `:max_by_time` scope could be more simply named `:newest`.
The `:created_by` attribute alias should be moved adjacent to the `:user` relation at the top.